src/streamlink_cli/main.py

```diff
@@ -125,6 +129,18 @@ def create_http_server(host=None, port=0):
 
     return http
 
+def create_title(plugin=None):
+    if args.title and plugin:
+        title = LazyFormatter.format(
+            maybe_encode(args.title),
+            title=lambda: plugin.get_title() or DEFAULT_STREAM_METADATA["title"],
+            author=lambda: plugin.get_author() or DEFAULT_STREAM_METADATA["author"],
+            category=lambda: plugin.get_category() or DEFAULT_STREAM_METADATA["category"],
+            game=lambda: plugin.get_category() or DEFAULT_STREAM_METADATA["game"]
+        )
+    else:
+        title = args.url
```

Comment:
Oh, I see what you're saying.

I thought args.url would only have the url opened when I wrote that, so to conform with that expected behavior, you would like an args.url of `hls://https://nasa-i.akamaihd.net/hls/live/253565/NASA-NTV1-Public/master.m3u8 name_key='bitrate'` to be converted to `hls://https://nasa-i.akamaihd.net/hls/live/253565/NASA-NTV1-Public/master.m3u8`.

Which you say should be just doing `title = args.url.split(' ')[0]`

Reply:
Doing that might not be acceptable (URLs can have spaces in them if they're not converted to %20, although), so if this is considered a blocker, we can just use the current behavior of changing the title to " - ".

I just thought args.url would be more helpful than " - ".

Reply:
Possibly the better fix is to rewrite the use of `args.url`to not put those params in it. I would want that done as a separate pull request, by someone who is not me. So for this PR, it should remain `args.url` or be " - ".

Reply:
1. I rethinked it, the builtin plugin will never be called without `--title`
so it will always be `args.url` not `self.title`
`plugin.url` could be used but it won't do any big impact, 
 because `self.url` is changed very rarely

2. maybe it is better not to use a default title,
   some third party apps might break (already uses some sort of own `--title`)

3. something else could be used instead of `DEFAULT_STREAM_METADATA["title"]`,
   maybe `plugin.url` so it won't show up as Unknown

this is my opinion, maybe there are also some other solutions.

Reply:
2. My opinion is that software needs to be free to make changes that make it better, without worrying about hypothetical 3rd party software incompatibilities. If it's a blocker, then I don't care that much, but the URL of the stream is a much more useful default than " - " in my opinion.

3. Why? If the title is unknown, then it's accurate. It indicates that the plugin used does not have a title getting method in it. The title is not the URL. This will just lead to unwanted behavior by users who were pleased by the behavior of getting the URL when using {title} and then being upset that it's changed once the feature is implemented for their plugin. Possibly you want a `{url}` format code, but I don't want to implement anything else on this PR that could block it.

Also, I'm going on vacation now until August. I'll still accept PRs but won't be working on this at least until then.

Reply:
> If it's a blocker, then I don't care that much,

it is not really a blocker, just some opinion from me it can be used like it is.

> DEFAULT_STREAM_METADATA["title"]

true, I did not think about the `"go away" once someone implements the feature for the plugin.`